src/lib/repositories/impl_v2/oauth2_repository_impl.py

```diff
@@ -82,13 +98,31 @@ def _login(
             refresh_token = build_user_credential_refresh_token(
                 user, client, self.env_config.oauth2_secret_key, client_scopes
             )
+            token_decoded = jwt.decode(
```

Comment:
The code is almost the same one as the above, could you optimize this?

Reply:
done.